src/assets/wise5/classroomMonitor/classroomMonitorComponents/grading-edit-component-max-score/grading-edit-component-max-score.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, Input, ViewEncapsulation } from '@angular/core';
+import { Subject, Subscription } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+import { TeacherProjectService } from '../../../services/teacherProjectService';
+import { UtilService } from '../../../services/utilService';
+
+@Component({
+  selector: 'grading-edit-component-max-score',
+  styles: ['.mat-form-field-infix { width: inherit; }'],
+  templateUrl: 'grading-edit-component-max-score.component.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class GradingEditComponentMaxScoreComponent {
+  @Input()
+  componentId: string;
+
+  @Input()
+  disabled: boolean;
+
+  @Input()
+  maxScore: number;
+
+  maxScoreChanged: Subject<string> = new Subject<string>();
+
+  @Input()
+  nodeId: string;
+
+  subscriptions: Subscription = new Subscription();
+
+  constructor(private ProjectService: TeacherProjectService, private UtilService: UtilService) {}
+
+  ngOnInit() {
+    this.maxScore = this.ProjectService.getMaxScoreForComponent(this.nodeId, this.componentId) || 0;
+    this.subscriptions.add(
+      this.maxScoreChanged.pipe(debounceTime(1000), distinctUntilChanged()).subscribe(() => {
+        this.saveMaxScore();
+      })
+    );
+  }
+
+  ngOnDestroy() {
+    this.subscriptions.unsubscribe();
+  }
+
+  saveMaxScore() {
+    const maxScore = this.UtilService.convertStringToNumber(this.maxScore);
+    if (typeof maxScore === 'number' && maxScore >= 0) {
```

Comment:
Is the typeof maxScore === 'number' check required?

Reply:
If you delete the max score in the input, this.maxScore becomes null, and maxScore >= 0 evaluates to true. UtilService.setMaxScoreForComponent does a null-check and won't save it, so perhaps it is redundant?

Reply:
Removed unnecessary checks.